syntax = "proto2";

package archive_box;

import "dtsdb/schema.proto";

message StoredDataId {
    optional string schema = 1;
    optional string blob = 2;
}

message Collection {
    option (dtsdb.table).name = "Collection";
    required string id = 1 [(dtsdb.field).is_id = true];
    optional string display_name = 2 [(dtsdb.field).merge = "latest"];
}

message Document {
    option (dtsdb.table).name = "Document";
    required string id = 1 [(dtsdb.field).is_id = true];
    optional StoredDataId data_id = 2;

    optional string display_name = 3 [(dtsdb.field).merge = "latest"];
    repeated string tags = 4 [(dtsdb.field).merge = "set_union"];
    map<string, string> metadata = 5 [(dtsdb.field).merge = "union_latest"];
}
